src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { NewCommentForm } from '../NewCommentForm';
+import './PostDetails.scss';
+import { Post, Comment } from '../../types';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
+
+type Props = {
+  postId: number | undefined;
+};
+
+export const PostDetails: React.FC<Props> = ({
+                                               postId,
+                                             }) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
+  const [visiblecomments, setVisiblecomments] = useState(false);
+  const [postDetails, setPostDetails] = useState<Post>();
+
+  const findPost = async () => {
+    if (postId) {
+      const result = await getPostbyId(postId);
+
+      setPostDetails(result);
+    }
+  };
+
+  useEffect(() => {
+    findPost();
```

Comment:
Since `findPost` is only used inside this useEffect, we can simply remove it from te component's root and move it to the `useEffect` callback.